tests/toolbox_image.sh

```diff
@@ -159,6 +228,26 @@ test_system_update_entrypoint()
     test -x "${rootfs_dir}${SYSTEM_UPDATE_ENTRYPOINT}"
 }
 
+test_execute_resize_partition_shrink_spool_ok()
+{
+    partition_table_file="/tmp/partition_table"
+    chroot "${rootfs_dir}" /sbin/sfdisk -d "${STORAGE_DEVICE_FILE}" > "${rootfs_dir}${partition_table_file}"
+
+    spool_size="$(( SECTOR_TOTAL - SPOOL_START ))"
+    new_spool_size="$(( spool_size / 2 ))"
+    new_spool_start="$(( SPOOL_START + new_spool_size ))"
```

Comment:
while completely relevant here, this is a note to ourselves, the final 'actual' partition table will need to account for partition and file-system alignment to flash erase sizes!